### PR TITLE
Rename EngineerRepair(able) to InstantlyRepair(able) and fix Dune 2000 engineers

### PR DESCRIPTION
**Pull request by Mailaender**
`EngineerRepairable` and `RepairBuilding` were already too narrow and C&C focussed so this just renames them.

I also discovered by accident that Dune 2000 engineers can repair but buildings won't let them.

### CODE REVIEW

File: mods/d2k/rules/defaults.yaml [hunk 1]
```diff
@@ -455,6 +455,7 @@
 		RepairingNotification: Repairing
 	SpawnActorsOnSell:
 		ActorTypes: light_inf
+	InstantlyRepairable:
```
Comment (abcdefg30) [inline mods/d2k/rules/defaults.yaml]: We want to disable this on the `sietch` actor then which inherits `^Building`.

**Comment by MustaphaTR** [comment]
Engineer repairing wasn't a thing in original D2k, are we sure we want to add it back? I think it was removed along with the changes to make the mod closer to original, but i guess not properly, if the trait was left on the engis.

**Comment by abcdefg30** [comment]
Needs a rebase.

File: OpenRA.Mods.Common/Traits/InstantlyRepairable.cs [hunk 2]
```diff
@@ -13,20 +13,20 @@
...
-	public class EngineerRepairType { }
+	public class InstantlyRepairType { }
 
 	[Desc("Eligible for instant repair.")]
-	public class EngineerRepairableInfo : ConditionalTraitInfo
+	public class InstantlyRepairableInfo : ConditionalTraitInfo
 	{
 		[Desc("Actors with these Types under EngineerRepair trait can repair me.")]
-		public readonly BitSet<EngineerRepairType> Types = default;
+		public readonly BitSet<InstantlyRepairType> Types = default(BitSet<InstantlyRepairType>);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/InstantlyRepairable.cs]: Seems like this fails with a lint error now?
```
Error: /home/runner/work/OpenRA/OpenRA/OpenRA.Mods.Common/Traits/InstantlyRepairable.cs(22,55): error IDE0034: 'default' expression can be simplified [/home/runner/work/OpenRA/OpenRA/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]
Error: /home/runner/work/OpenRA/OpenRA/OpenRA.Mods.Common/Traits/InstantlyRepairs.cs(24,55): error IDE0034: 'default' expression can be simplified [/home/runner/work/OpenRA/OpenRA/OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]
```
I guess this means this should be
```suggestion
		public readonly BitSet<InstantlyRepairType> Types = default;
```

File: OpenRA.Mods.Common/Traits/InstantlyRepairs.cs [hunk 3]
```diff
@@ -18,10 +18,10 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Can instantly repair other actors, but gets consumed afterwards.")]
-	public class EngineerRepairInfo : ConditionalTraitInfo
+	public class InstantlyRepairsInfo : ConditionalTraitInfo
 	{
-		[Desc("Uses the \"EngineerRepairable\" trait to determine repairability.")]
-		public readonly BitSet<EngineerRepairType> Types = default;
+		[Desc("Uses the " + nameof(InstantlyRepairable) + " trait to determine repairability.")]
+		public readonly BitSet<InstantlyRepairType> Types = default(BitSet<InstantlyRepairType>);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/InstantlyRepairs.cs]: Same here.

File: OpenRA.Mods.Common/Traits/InstantlyRepairable.cs (hunk 2 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/InstantlyRepairable.cs]: Rebase problems.

File: OpenRA.Mods.Common/UpdateRules/Rules/20210321/RenameEngineerRepair.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UpdateRules/Rules/20210321/RenameEngineerRepair.cs]: This needs an update now too.

File: mods/cnc/rules/defaults.yaml [hunk 5]
```diff
@@ -744,8 +744,12 @@
 		AreaTypes: building
 	SpawnActorsOnSell:
 		ActorTypes: e6,e1,e1,e1
+<<<<<<< HEAD
```
Comment (abcdefg30) [inline mods/cnc/rules/defaults.yaml]: This is probably unintentional.

**Comment by abcdefg30** [comment]
Seems like this has a conflict again. :/

**Review by penev92 — APPROVED** [review]
OK

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b6fbf943559f655cce9d9f5684777268af30b8dd...f08c914049a458d4ee2995c52235dfa97483339c)

File: OpenRA.Mods.Common/UpdateRules/Rules/20210321/RenameEngineerRepair.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+﻿#region Copyright & License Information
```
Comment (PunkPun) [inline OpenRA.Mods.Common/UpdateRules/Rules/20210321/RenameEngineerRepair.cs:1]: Wrong folder

**Pull request merged**